### PR TITLE
Remove GraphCall

### PR DESCRIPTION
**Pull request by wschin**
Most of the time, we only have one GraphCall inside the training
graph for calling `forward` of the model once. Thus, it seems more
natural to automatically add that `forward` into training graph.
This PR changes the IR for training and change training graph to
a partial graph. The full training graph is formed by merging
(partial) training graph and the inference graph. In addition,
GraphCall is removed because it will be rarely used.

### CODE REVIEW

File: onnx/onnx-ml.proto [hunk 1]
```diff
@@ -242,20 +242,23 @@ message TrainingInfoProto {
...
   // it computes outputs to update initializers in its own or inference graph's
-  // initializer lists. In general, this graph contains loss node, gradient node,
-  // optimizer node, increment of iteration count, and some calls to the inference
-  // graph.
+  // initializer lists. In general, this field contains loss node, gradient node,
+  // optimizer node, increment of iteration count.
   //
-  // The field algorithm.node is the only place the user can use GraphCall
-  // operator. The only callable graph is the one stored in ModelProto.graph.
+  // In runtime, the training graph, "algorithm", is always combined with
```
Comment (gramalingam) [inline onnx/onnx-ml.proto:253]: Maybe add/change as follows?
```
An execution of the training step is performed by executing the graph obtained by combining
the inference graph (namely "ModelProto.graph") and the algorithm graph. That is, the actual ...
```

File: onnx/onnx-ml.proto [hunk 2]
```diff
@@ -242,20 +242,23 @@ message TrainingInfoProto {
...
+  // the inference graph, "ModelProto.graph". That is, the actual
+  // input/initializer/output/node/value_info/sparse_initializer list of
+  // training graph is the concatenation of
+  // "ModelProto.graph.input/initializer/output/node/value_info/sparse_initializer"
+  // and "algorithm.input/initializer/output/node/value_info/sparse_initializer".
```
Comment (gramalingam) [inline onnx/onnx-ml.proto:258]: Add "in that order. This combined graph must satisfy the normal ONNX conditions. For example, an input of a node in the algorithm graph may reference the output of a node in the inference graph (but not the other way round)."

File: onnx/onnx-ml.proto (hunk 1 above)
Comment (wschin) [reply onnx/onnx-ml.proto:253]: Ok. Thanks.

File: onnx/onnx-ml.proto3 [hunk 3]
```diff
@@ -242,20 +242,28 @@ message TrainingInfoProto {
...
-  // optimizer node, increment of iteration count, and some calls to the inference
-  // graph.
+  // initializer lists. In general, this field contains loss node, gradient node,
+  // optimizer node, increment of iteration count.
   //
-  // The field algorithm.node is the only place the user can use GraphCall
-  // operator. The only callable graph is the one stored in ModelProto.graph.
+  // An execution of the training algorithm step is performed by executing the
+  // graph obtained by combining the inference graph (namely "ModelProto.graph")
+  // and the "algorithm" graph. That is, the actual the actual
```
Comment (wschin) [inline onnx/onnx-ml.proto3:255]:
```suggestion
  // and the "algorithm" graph. That is, the actual
```

File: onnx/onnx-ml.proto3 [hunk 4]
```diff
@@ -242,20 +242,28 @@ message TrainingInfoProto {
...
+  // input/initializer/output/node/value_info/sparse_initializer list of
+  // the training graph is the concatenation of
+  // "ModelProto.graph.input/initializer/output/node/value_info/sparse_initializer"
+  // and "algorithm.input/initializer/output/node/value_info/sparse_initializer"
+  // in that order. This combined graph must satisfy the normal ONNX conditions.
+  // For example, an input of a node in the "algorithm" graph may reference the
+  // output of a node in the inference graph (but not the other way round).
```
Comment (wschin) [inline onnx/onnx-ml.proto3:262]: Maybe add some more comments.

Inf:
	ModelProto.grpah
		initializer_w -> MatMul1 -> tensor_x_inf

Algorithm:
	ModelProto.training_info[i].algorithm
		initializer_w -> GraphCall(MatMul1) -> tensor_x -> MatMul2 -> tensor_y -> MatMul3 -> tensor_z


Inf:
	ModelProto.grpah
                initializer_w -> MatMul1 -> tensor_x

Algorithm:
	ModelProto.training_info[i].algorithm
                       tensor_x -> MatMul2 -> tensor_y -> MatMul3 -> new_initializer_w
	ModelProto.training_info[i].update_binding
		       "initializer_w", "new_initializer_w"

Underlying graph represented by TrainingInfoProto.algorithm (with the context of inference graph)
 GraphProto.input = concat(ModelProto.input, algorithm.input)
 GraphProto.node = concat(ModelProto.node, algorithm.node)
 GraphProto.output = concat(ModelProto.output, algorithm.output)

**Pull request merged**